ttnn/cpp/ttnn/operations/conv/conv2d/conv2d_utils.cpp

```diff
@@ -397,19 +402,85 @@ bool use_matmul_for_1x1_conv(
            padding[1] == 0 && dilation[0] == 1 && dilation[1] == 1 && groups == 1;
 }
 
+
 // Implements a heuristic for selecting shard layout based on how many tenix cores are available
 // for each shard.
 static TensorMemoryLayout select_shard_spec(
+    const Conv2dConfig& conv_config,
+    const DeviceComputeKernelConfig& compute_config,
     bool is_mm_conv,
     uint32_t batch_size,
     uint32_t in_channels,
     uint32_t out_channels,
     uint32_t output_height,
     uint32_t output_width,
     uint32_t weights_width,
+    uint32_t input_height,
     uint32_t input_width,
-    ShardOrientation shard_orientation,
+    uint32_t groups,
+    std::array<uint32_t, 2> kernel_size,
     const CoreCoord& compute_grid_size) {
+    auto shard_orientation = conv_config.transpose_shards ? ShardOrientation::COL_MAJOR : ShardOrientation::ROW_MAJOR;
+
+    auto get_conv2d_resources_for_shard_scheme = [&](TensorMemoryLayout shard_layout)->std::tuple<uint32_t, uint32_t, uint32_t> {
```

Comment:
this doesn't seem to be used anywhere?

Reply:
This PR is only for implementing the infrastructure to get the L1 usage.

Reply:
I'm not following you have added unused code

Reply:
Remove this code.

Reply:
Done.